BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -5681,7 +5681,10 @@ public override string GetInfo()
                             output.AppendLine($"Cannister Round");
                             output.AppendLine($" - Submunition count: {binfo.subProjectileCount}");
                         }
-                        output.AppendLine($"Estimated Penetration: {ProjectileUtils.CalculatePenetration(binfo.caliber, binfo.bulletVelocity, binfo.bulletMass, binfo.apBulletMod, 940, 8.45001135e-07f, 0.656060636f, 1.20190930f, 1.77791929f):F2} mm");
+                        if (binfo.explosive == "shaped")
+                            output.AppendLine($"Estimated Penetration: {ProjectileUtils.CalculatePenetration(binfo.caliber > 0 ? binfo.caliber * 0.05f : 6f, 5000f, binfo.tntMass * 0.0555f, binfo.apBulletMod, 940, 0.00000094776185184f, 0.6560606203f, 1.201909309f, 1.777919321f):F2} mm");
+                        else
```

Comment:
Might work better to have the shaped charge penetration be a separate line (" - Shaped Charge Penetration"?), on the off chance that someone makes custom bullets with a delay/penetrating fuzed shaped charge. Incredibly niche, and for most usage (e.g. barring naval shells(for which SC warheads would be of dubious usage anyway)) probably not going to add more than maybe 100mm penetration, so not a big deal either way, but something to consider.

Reply:
I'll make that change, good call.